src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -103,12 +110,27 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
           logger.info("Triggering Jenkins build on WorkflowJob " + wj.getFullName() + " due to running on kubernetes due to " + causeDescription(actions));
           return true;
         }
-        logger.info("Triggering OpenShift Build for WorkflowJob " + wj.getFullName() + " due to " + causeDescription(actions));
+
+        /* We are always finding the buildConfig by the the name of JenkinsJobName
+         This is failing in the case of repoName and buildConfigName and triggering a build by pushing to master
+         not same we need to find the buildConfig by gitLabel also as added in launcher backend and
+         we do in PipelineJobListener in case of initial creation
+         */
+        BuildConfig foundBuildConfig = OpenShiftUtils.getBuildConfig(namespace, buildConfigname);
+
+        /*In this case if buildConfig name is not same as jenkins jobName converted to buildConfigName
```

Comment:
Let's try to get rid of the comment. A better way could refactor the code by wrapping build request logic into a private method `triggerBuildRequest()`

_let method name says what this piece of code is doing_